src/tribol/mesh/CouplingScheme.cpp

```diff
@@ -1546,122 +1601,124 @@ void CouplingScheme::computeCommonPlaneTimeStep(real &dt)
+
+        if (dim == 3)
+        {
+          proj_delta_z1 = plane.m_cZf1 + dt * vel_f1[2] - plane.m_cZf2;
+          proj_delta_z2 = plane.m_cZf2 + dt * vel_f2[2] - plane.m_cZf1;
+
+          proj_delta_n_1 += proj_delta_z1 * fn2[2];
+          proj_delta_n_2 += proj_delta_z2 * fn1[2];
+        }
+
+        // Reset the dt velocity check only for faces with continued interpen that exceeds the 
+        // max allowable gap AND where the current gap did NOT exceed that face's max allowable
+        // gap per check 1 (would result in same dt calc).
+        //
+        // Note:
+        // If proj_delta_n_i < 0, (i=1,2) there is interpen from the velocity projection. 
+        // Check this interpen against the maximum allowable to determine if a velocity projection 
+        // timestep estimate is still required.
+        if (dt1_vel_check && !dt1_check1) // continued interpen
+        {
+          dt1_vel_check = (proj_delta_n_1 < 0.) ? ((std::abs(proj_delta_n_1) > max_delta1) ? true : false) : false;
+        }
+        
+        if (dt2_vel_check && !dt2_check1) // continued interpen
+        {
+          dt2_vel_check = (proj_delta_n_2 < 0.) ? ((std::abs(proj_delta_n_2) > max_delta2) ? true : false) : false;
+        }
+
+        // compute velocity projection based dt (check 2) using a RESET gap (g=0) such that
+        // the velocity projected gap does not exceed the max allowable gap. This avoid timestep
+        // crashes for velocity projected gaps slightly in excess of the max allowable and still
+        // allows for a soft contact response without a timestep crash.
+        //
+        // v1_dot_n1 > 0 and v2_dot_n2 > 0 for further interpen
+        dt1 = (dt1_vel_check) ? alpha * max_delta1 / v1_dot_n1 : dt1;
+        dt2 = (dt2_vel_check) ? alpha * max_delta2 / v2_dot_n2 : dt2; 
+
+        // Keep debug print statements. This routine is still in the testing phase
+        //std::cout << "dt1_vel_check, (proj_delta_n_1+max_delta1), v1_dot_n1: " << dt1_vel_check << ", " 
+        //          << proj_delta_n_1+max_delta1 << ", " << v1_dot_n1 << std::endl;
+        //std::cout << "dt2_vel_check, (proj_delta_n_2+max_delta2), v2_dot_n2: " << dt2_vel_check << ", " 
+        //          << proj_delta_n_2+max_delta2 << ", " << v2_dot_n2 << std::endl;
+        //std::cout << "dt1 and dt2: " << dt1 << ", " << dt2 << std::endl;
+
+        // update dt_temp2 only for positive dt1 and/or dt2
+        if (dt1 > 0.)
+        {
+          RAJA::atomicMin<RAJA::auto_atomic>( &dt_temp[1],
+                                              axom::utilities::min(dt1, 1.e6) );
+        }
+        if (dt2 > 0.)
+        {
+          RAJA::atomicMin<RAJA::auto_atomic>( &dt_temp[1],
+                                              axom::utilities::min(1.e6, dt2) );
+        }
+        if (dt1 < 0. || dt2 < 0.)
+        {
+          msg[3] = true;
+        }
 
       } // end check 2
-
-      ++cpID;
```

Comment:
I guess we don't need this counter since we aren't indexing into the contact plane manager anymore?

Reply:
We're looping over the contact planes directly instead of interface pairs.  This is to make sure all of the threads are doing work on an active contact plane.